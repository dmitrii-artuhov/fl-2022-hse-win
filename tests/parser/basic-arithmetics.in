start=🤯expr🤯
🤯expr🤯 👉 🤯expr🤯 🥵+🥵 🤯term🤯 🤌 🤯expr🤯 🥵-🥵 🤯term🤯 🤌 🤯term🤯🗿
🤯term🤯 👉 🤯term🤯 🥵*🥵 🤯mult🤯 🤌 🤯term🤯 🥵/🥵 🤯mult🤯 🤌 🤯mult🤯🗿
🤯mult🤯 👉 🥵(🥵 🤯expr🤯 🥵)🥵   🤌 🥵0🥵 🤌 🥵1🥵 🤌 🥵2🥵 🤌 🥵3🥵🗿