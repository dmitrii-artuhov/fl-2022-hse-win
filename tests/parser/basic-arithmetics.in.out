
-- Grammar --
Terminals: ['(', ')', '*', '+', '-', '/', '0', '1', '2', '3']
Non-terminals: ['expr', 'mult', 'term']
Start: 'expr'
Rules:
'expr' -> ['expr', '+', 'term'] | ['expr', '-', 'term'] | ['term']
'term' -> ['term', '*', 'mult'] | ['term', '/', 'mult'] | ['mult']
'mult' -> ['(', 'expr', ')'] | ['0'] | ['1'] | ['2'] | ['3']

Chomsky normal form: False
        
